app/controllers/staff/actuals_controller.rb

```diff
@@ -22,45 +24,45 @@ def new
   def create
     @activity = activity
     authorize @activity
+    @actual = ActualForm.new(actual_params.merge({parent_activity: @activity}))
 
-    result = CreateActual.new(activity: @activity)
-      .call(attributes: actual_params)
-    @actual = result.object
+    actual_created = @actual.valid? && CreateActual.new(
+      activity: @activity,
+      user: current_user,
+      report: Report.editable_for_activity(@activity)
+    ).call(attributes: @actual.attributes).success?
 
-    if result.success?
-      flash[:notice] = t("action.actual.create.success")
-      redirect_to organisation_activity_path(@activity.organisation, @activity)
-    else
-      render :new
-    end
+    return render :new unless actual_created
+
+    flash[:notice] = t("action.actual.create.success")
+    redirect_to organisation_activity_path(@activity.organisation, @activity)
   end
 
   def edit
-    @actual = Actual.find(id)
-    authorize @actual
-
     @activity = Activity.find(activity_id)
+    @actual = ActualForm.new(attributes_for_editing)
+
+    authorize(@actual, policy_class: ActualPolicy)
 
     prepare_default_activity_trail(@activity)
-    add_breadcrumb t("breadcrumb.actual.edit"), edit_activity_actual_path(@activity, @actual)
+    add_breadcrumb t("breadcrumb.actual.edit"), edit_activity_actual_path(@activity, @actual.id)
```

Comment:
why do we need to add .id here?

Reply:
Because `@actual` isn't an instance of `Actual`, it's an `ActualForm`.

Reply:
🙄  ok!